cmd/argoexec/commands/artifact/delete.go

```diff
@@ -52,63 +53,121 @@ func NewArtifactDeleteCommand() *cobra.Command {
 	}
 }
 
-func deleteArtifacts(labelSelector string, ctx context.Context, artifactGCTaskInterface wfv1alpha1.WorkflowArtifactGCTaskInterface) error {
+type request struct {
+	Task             *v1alpha1.WorkflowArtifactGCTask
+	NodeName         string
+	ArtifactNodeSpec *v1alpha1.ArtifactNodeSpec
+}
+
+type response struct {
+	Task     *v1alpha1.WorkflowArtifactGCTask
+	NodeName string
+	Results  map[string]v1alpha1.ArtifactResult
+	Err      error
+}
 
+func deleteArtifacts(labelSelector string, ctx context.Context, artifactGCTaskInterface wfv1alpha1.WorkflowArtifactGCTaskInterface) error {
 	taskList, err := artifactGCTaskInterface.List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
 	if err != nil {
 		return err
 	}
+	taskWorkers := env.LookupEnvIntOr(common.EnvExecGCWorkers, 4)
```

Comment:
Should we look into how many cores are available and base the parallelism off that ? 
I can imagine some deployments being backed by some heavy servers. 


```python
if log2(cores) + 4 >= cores:
  return cores
else:
  return log2(cores) + 4
```

Something like the above growth rate could make sense because it doesn't scale with cores linearly. 

Not even sure if this is good idea, just a suggestion.

Reply:
This is deployed as a separate pod, so users can tune resource limits and `ARGO_EXEC_GC_WORKERS` to fit. This is a heavily IO bound process as we're mostly just dispatching deletes to a remote blob store, and it's dependent on how that is configured how much it will benefit from parallelism, so attempting an autotune is probably not worthwhile.

Reply:
Makes sense, yeah I can imagine this being majorly bound on the network. Makes sense that an autotune isn't worthwhile.